### PR TITLE
Implement storeValidationRecordIfNecessary for JITaaS

### PR DESCRIPTION
**Pull request by dmitry-ten**
[skip ci]

Signed-off-by: Dmitry Ten <Dmitry.Ten@ibm.com>

### CODE REVIEW

**Comment by dmitry-ten** [comment]
@mpirvu could you please review?

File: runtime/compiler/env/j9methodServer.cpp [hunk 1]
```diff
@@ -1355,3 +1355,111 @@ TR_ResolvedJ9JITaaSServerMethod::getCachedResolvedMethod(TR_ResolvedMethodKey ke
...
+   {
+   TR_J9VMBase *fej9 = (TR_J9VMBase *) comp->fe();
+
+   bool storeClassInfo = true;
+   bool fieldInfoCanBeUsed = false;
+   TR_AOTStats *aotStats = ((TR_JitPrivateConfig *)fej9->_jitConfig->privateConfig)->aotStats;
+   bool isStatic = false;
+
+   TR::CompilationInfo *compInfo = TR::CompilationInfo::get(fej9->_jitConfig);
+   TR_RelocationRuntime *reloRuntime = compInfo->getCompInfoForThread(fej9->vmThread())->reloRuntime();
```
Comment (mpirvu) [inline runtime/compiler/env/j9methodServer.cpp]: We should make sure that the reloRuntime stored in compInfo is of type TR_SharedCacheJITaaSRelocationRuntime

File: runtime/compiler/env/j9methodServer.cpp [hunk 2]
```diff
@@ -1355,3 +1355,111 @@ TR_ResolvedJ9JITaaSServerMethod::getCachedResolvedMethod(TR_ResolvedMethodKey ke
...
+
+   isStatic = (reloKind == TR_ValidateStaticField);
+
+   TR_J9ServerVM *feServer = (TR_J9ServerVM *) fej9;
+   J9Class *clazz = (J9Class *) feServer->TR_J9ServerVM::getClassOfMethod((TR_OpaqueMethodBlock *) ramMethod);
```
Comment (mpirvu) [inline runtime/compiler/env/j9methodServer.cpp]: Would it be bad if we called directly feServer->getClassOfMethod()?

Comment (dmitry-ten) [reply runtime/compiler/env/j9methodServer.cpp]: If I'm not wrong, `fej9` will be instantiated to `TR_SharedCacheServerVM`, so calling `feServer->getClassOfMethod()` will call the AOT of the method, which we do not want.

File: runtime/compiler/env/j9methodServer.cpp (hunk 1 above)
Comment (dmitry-ten) [reply runtime/compiler/env/j9methodServer.cpp]: Yeah, but that should probably go in a different commit, the one that implements RelocationRuntime.

File: runtime/compiler/env/j9methodServer.cpp (hunk 2 above)
Comment (mpirvu) [reply runtime/compiler/env/j9methodServer.cpp]: I agree

**Pull request merged**